modules/core/src/main/scala/org/tessellation/domain/rewards/EpochRewards.scala

```diff
@@ -0,0 +1,12 @@
+package org.tessellation.domain.rewards
+
+import scala.collection.immutable.SortedMap
+
+import org.tessellation.schema.GlobalIncrementalSnapshot
+import org.tessellation.schema.balance.Amount
+import org.tessellation.schema.epoch.EpochProgress
+import org.tessellation.sdk.domain.rewards.Rewards
+
+trait EpochRewards[F[_]] extends Rewards[F, GlobalIncrementalSnapshot] {
```

Comment:
@kpudlik ideally we should have the definition in `domain` (models (case classes), programs, storage and service traits, etc.) and implementations in `infrastructure`

Reply:
Here the case is more complex. The base `Rewards` trait should be in `sdk` because it is meant to be reused and implemented by custom Currency.
On the other hand `EpochRewards` implementation is `GlobalSnapshot` specific. So in our case trait would be in `sdk/domain` and implementatation in `core/infrastructure`. According to "Please stop splitting interfaces that have single implementation into different packages." everything should be in sdk 🤔